packages/frontend/src/build/createApi.ts

```diff
@@ -73,17 +73,31 @@ export function outputCharts(
   }
 }
 
-function getActivityChart(
+export function getActivityChart(
   apiChart: ActivityApiChart,
   ethereumChart: ActivityApiChart,
 ): FrontendActivityChart {
-  const length = Math.min(apiChart.data.length, ethereumChart.data.length)
+  const lastProjectTimestamp = apiChart.data.at(-1)?.[0]
+  if (!lastProjectTimestamp) {
+    throw new Error('No data in activity chart')
+  }
+  const ethChartTimestampIndex = ethereumChart.data.findIndex(
+    (x) => x[0].toNumber() === lastProjectTimestamp.toNumber(),
+  )
+  if (ethChartTimestampIndex === -1) {
+    throw new Error('No matching timestamp in ethereum chart')
+  }
+  const alignedEthChartData = ethereumChart.data.slice(
+    0,
+    ethChartTimestampIndex + 1,
+  )
+  const length = Math.min(apiChart.data.length, alignedEthChartData.length)
   return {
```

Comment:
this function would be much more readable if it only said: cut the data and return, write everything above return as a seperate function